src/Symfony/Component/Stream/FilesystemStreamWrapper.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Stream;
+
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Native filesystem stream wrapper.
+ *
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ *
+ * @internal
+ */
+class FilesystemStreamWrapper implements StreamWrapperInterface
+{
+    /** @var resource|null */
+    public $context;
+
+    private $baseDir;
+    private $filesystem;
+    private $directory;
+
+    public function __construct(string $baseDir, Filesystem $filesystem = null)
```

Comment:
yeah this doesnt make sense yet :P the class is registered by name, so making this service-aware requires some thought still.